### ISSUE TITLE
Class 'LookupAttributeMock' incorrectly implements interface 'LookupAttribute'.

### ISSUE THREAD

**Issue by jorgeolive** [camelCaseDave/xrm-mock#37]
Hello,

When compiling the ts files I am getting the following errors:

node_modules/xrm-mock/dist/xrm-mock/attributes/lookupattribute/lookupattribute.mock.d.ts:4:22 - error TS2420: Class 'LookupAttributeMock' incorrectly implements interface 'LookupAttribute'.
  Types of property 'controls' are incompatible.
    Type 'ItemCollectionMock<LookupControlMock>' is not assignable to type 'ItemCollection<LookupControl>'.
      Types of property 'forEach' are incompatible.
        Type '{ (delegate: () => void): void; (delegate: IterativeDelegate<LookupControlMock>): void; }' is not assignable to type '(delegate: IterativeDelegate<LookupControl>) => void'.
          Types of parameters 'delegate' and 'delegate' are incompatible.

4 export declare class LookupAttributeMock extends AttributeMock<LookupControlMock, Xrm.LookupValue[]> implements Xrm.Attributes.LookupAttribute {
                       ~~~~~~~~~~~~~~~~~~~

node_modules/xrm-mock/dist/xrm-mock/controls/lookupcontrol/lookupcontrol.mock.d.ts:16:5 - error TS2416: Property 'removePreSearch' in type 'LookupControlMock'
is not assignable to the same property in base type 'LookupControl'.
  Type '(handler: () => void) => void' is not assignable to type '(handler: ContextSensitiveHandler) => void'.
    Types of parameters 'handler' and 'handler' are incompatible.

16     removePreSearch(handler: () => void): void;
       ~~~~~~~~~~~~~~~

node_modules/xrm-mock/dist/xrm-mock/controls/tab/tab.mock.d.ts:2:22 - error TS2420: Class 'TabMock' incorrectly implements interface 'Tab'.
  Type 'TabMock' is missing the following properties from type 'Tab': addTabStateChange, removeTabStateChange

2 export declare class TabMock implements Xrm.Controls.Tab {
                       ~~~~~~~

node_modules/xrm-mock/dist/xrm-mock/webapi/webapi.mock.d.ts:8:5 - error TS2416: Property 'createRecord' in type 'WebApiMock' is not assignable to the same property in base type 'WebApi'.
  Type '(entityLogicalName: string, record: any) => PromiseLike<string>' is not assignable to type '(entityLogicalName: string, record: any) => PromiseLike<CreateResponse>'.
    Type 'PromiseLike<string>' is not assignable to type 'PromiseLike<CreateResponse>'.
      Type 'string' is not assignable to type 'CreateResponse'.

8     createRecord(entityLogicalName: string, record: any): Xrm.Async.PromiseLike<string>;
      ~~~~~~~~~~~~

node_modules/xrm-mock/dist/xrm-mock/webapioffline/webapioffline.mock.d.ts:4:5 - error TS2416: Property 'createRecord' in type 'WebApiOfflineMock' is not assignable to the same property in base type 'WebApiOffline'.
  Type '(entityLogicalName: string, record: any) => PromiseLike<string>' is not assignable to type '(entityLogicalName: string, record: any) => PromiseLike<CreateResponse>'.
    Type 'PromiseLike<string>' is not assignable to type 'PromiseLike<CreateResponse>'.

4     createRecord(entityLogicalName: string, record: any): Xrm.Async.PromiseLike<string>;
      ~~~~~~~~~~~~

node_modules/xrm-mock/dist/xrm-mock/webapionline/webapionline.mock.d.ts:8:5 - error TS2416: Property 'createRecord' in type 'WebApiOnlineMock' is not assignable to the same property in base type 'WebApiOnline'.
  Type '(entityLogicalName: string, record: any) => PromiseLike<string>' is not assignable to type '(entityLogicalName: string, record: any) => PromiseLike<CreateResponse>'.
    Type 'PromiseLike<string>' is not assignable to type 'PromiseLike<CreateResponse>'.

8     createRecord(entityLogicalName: string, record: any): Xrm.Async.PromiseLike<string>;

Not sure if I am doing something incorrectly. Any help please?

Regards,
Jorge.

**Comment by jordimontana82** [comment]
Try building against @types/xrm v 9.0.7 for now ....

**Comment by camelCaseDave** [comment]
Should be fixed in xrm-mock 3.4.12 which builds against @types/xrm@9.0.7
